### PR TITLE
(SERVER-2123) Create job to run gatling against puppet branch

### PR DESCRIPTION
**Pull request by haus**
In order to be able to better test how some changes to puppet affect
performance, this commit adds a job that can run gatling against a
puppet branch or ref. It clones and checks out the ref locally and
rsyncs it on top of the puppet-agent installation, which is gross, but
ultimately far simpler and faster than crafting a vanagon pipeline.

### CODE REVIEW

**Comment by haus** [comment]
Please hold. Something is weird here.

File: jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy [hunk 1]
```diff
@@ -212,6 +212,14 @@ def step020_install_server(SKIP_SERVER_INSTALL, script_dir, server_era, agent_ve
     }
 }
 
+def step021_clone_and_rsync(script_dir, remote, ref) {
```
Comment (Magisus) [inline jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: Are there rules around this numbering scheme (here and the way we tend to do steps in the main CI Jenkins pipelines)? I've always been a little confused by it.

Comment (haus) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: No rules, it just mirrors the order in the pipeline itself. I figured right after the server install would be the best spot, but I don't care too much.

**Comment by haus** [comment]
The issue is with puppet itself, and only puppet post 5.4.0 (https://tickets.puppetlabs.com/browse/PUP-8489)

File: jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy (hunk 1 above)
Comment (Magisus) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: I was more curious about the intervals, when to increment 1 vs. 5 vs. 10. Always seems sorta arbitrary, which it very well might be :)

Comment (haus) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: Oh yea. Totally arbitrary.

Comment (haus) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: This *felt* closer to the install 🤷‍♀️

**Comment by lindboe** [comment]
Why did we go with adding a new type of scenario/job instead of making this a parameter on the original puppetserver-infinite job?

**Comment by haus** [comment]
mostly because parameters that might not be set are kind of a pain to deal with, but also last time the complexity of param vs jenkinsfile entry came up. I can certainly wedge this into the existing scenario if that seems better.

**Comment by lindboe** [comment]
I'm not sure I understand -- why wouldn't the parameters be set? We are providing a default right?

**Comment by haus** [comment]
maybe i misunderstood the question. i thought you were asking about the new pipeline definition. i think it would work fine merged into the other infinite job, as long as it starts using the new pipeline. i can squash it together.

**Comment by lindboe** [comment]
Oh yeah I guess I wasn't really distinguishing between pipeline/job. I was just thinking that since puppetserver-infinite was supposed to be able to take all the params, that's where I expected this change to go.

**Comment by haus** [comment]
@lindboe updated. Have a job running and will kick off a fresh one in a few minutes to validate the changes, but it looks good so far.

**Comment by haus** [comment]
Hmm. Looks like known_hosts fun times with github now. Update coming.

**Comment by haus** [comment]
Green run from this branch here: http://puppetserver-perf-driver55.delivery.puppetlabs.net:8080/job/puppetserver-infinite/48/console

**Pull request merged**